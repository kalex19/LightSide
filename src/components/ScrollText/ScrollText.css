.scrollText {
  display: flex;
  justify-content: center;
  height: 600px;
  perspective: 400px;
  font-family: 'Pathway Gothic One', sans-serif;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 200%;
  text-align: justify;
  font-size: 2em;
  color: white;
}

.crawl {
  position: relative;
  top: -100px;
  transform-origin: 50% 100%;
  position: relative;
  transform-origin: 50% 100%;
  animation: crawl 90s linear;
}

.filmInfo {
  font-size: 90%;
  text-align: center;
}

.filmInfo h1 {
  margin: 0 0 100px;
  text-transform: uppercase;
}

@keyframes crawl {
  0% {
    top: 0;
    transform: rotateX(20deg) translateZ(0);
  }

  100% {
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}