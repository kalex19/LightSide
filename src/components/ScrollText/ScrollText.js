import React from 'react';

export default function ScrollText(){
	return <div />;
}
