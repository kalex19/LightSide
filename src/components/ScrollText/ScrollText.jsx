import React, { Component } from 'react';
import './ScrollText.css';

export class ScrollText extends Component {
	constructor() {
		super();
		this.state = {
			film: {},
			error: ''
		};
	}

	componentDidMount() {
		let range = 7;
		let num = Math.floor(Math.random() * Math.floor(range)) + 1;
		let url = `https://swapi.co/api/films/${num}`;
		fetch(url).then(response => response.json()).then(data => this.setState({ film: data })).catch(error =>
			this.setState({
				error: error.message
			})
		);
	}
	render() {
		const { film } = this.state;
		return (
			<section className="scrollText">
				<div className="crawl">
					<p>{film.opening_crawl}</p>
					<div className="filmInfo">
						<p>{film.title}</p>
						<p>{film.release_date}</p>
					</div>
				</div>
			</section>
		);
	}
}

export default ScrollText;
