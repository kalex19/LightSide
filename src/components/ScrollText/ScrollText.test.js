import React from 'react';
import {
    shallow
} from 'enzyme';
import ScrollText from './ScrollText.jsx';

describe('ScrollText', () => {
    let wrapper;
    let mockFilm;


    beforeEach(() => {
        wrapper = shallow( < ScrollText / > );
        mockFilm = {
            name: 'movie 1',
            release_date: '12-12-12',
            opening_crawl: 'Star wars is the best'
        }
        window.fetch = jest.fn().mockImplementation(() => {
            return Promise.resolve({
                ok: true,
                json: () => Promise.resolve(mockFilm)
            })
        })
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call fetch with correct params', () => {
        const expected = (`https://swapi.co/api/films/12`);
        ScrollText(mockFilm)
        expect(window.fetch).toHaveBeenCalledWith(...expected);
    });

    it('should return a parsed response if status is ok', async () => {
        const result = await ScrollText(mockFilm);

        expect(result).toEqual(mockFilm);
    });

    it('should return an error if status is not ok', async () => {
        window.fetch = jest.fn().mockImplementationOnce(() => {
            return Promise.resolve({
                ok: false
            })
        });

        await expect(ScrollText()).rejects.toEqual(Error('Error fetching data'));
    });
})