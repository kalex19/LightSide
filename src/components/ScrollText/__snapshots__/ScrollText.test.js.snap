// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollText should match snapshot 1`] = `
<section
  className="scrollText"
>
  <div
    className="crawl"
  >
    <p />
    <div
      className="filmInfo"
    >
      <p />
      <p />
    </div>
  </div>
</section>
`;
